### PR TITLE
Fixes Issue #150

### LINKED ISSUE #150: Extend Onboarding Tour Across Multiple Pages After Last Step on map/page
**Original Issue** https://github.com/oss-slu/lrda_website/issues/138

**Description**
**As** a new user,
**I want** the on boarding tour to continue across multiple pages,
**So that** I can learn the functionality of the application seamlessly even when navigating between different pages.

Currently, the on boarding tour in the application is limited to the map/page, with tooltips explaining key features only on that page. This enhancement will extend the on boarding experience so that after the last tooltip on map/page, the user is automatically redirected to another page and the tour continues without interruption.

**Tasks**
- [x] Ensure that the tour resumes seamlessly on the next page, starting from the first step relevant to that page.
- [x] Track the tour's progress across pages to ensure it resumes correctly after redirection
- [x] Implement a smooth page transition when the tour redirects the user to the next page (the user should not need to restart the tour manually).
- [x] Use React Router (or the appropriate routing solution) to handle navigation between pages.
- [x] regression free
    - committed code will not break other functionalities or previous commits
- [ ] Create a test cases for your code 
    - [ ] Ensure that the tour starts correctly.
    - [ ] Ensure that users can skip or continue the tour seamlessly.
    - [ ] Ensure that the tour progresses step-by-step without issues.

**Acceptance Criteria**
- [x] After the user clicks "Next" on the final tooltip on the map/page, they are redirected to the next page.
- [x] Tour progress is correctly tracked, and the correct step resumes after redirection.
- [x] The "Skip Tour" button works across pages, allowing users to exit the tour at any time.
- [ ] Users can navigate back and forth between pages, and the tour should still function properly.
- [ ] Create at least one test case for you code
    - [ ] Unit test to check functionality.
    - [ ] Integration test to verify correct operation across components.
- [x] Code is well commented
- [x] Only commit changes that you physically typed.
- [x] Code should pass the test case before merge

Code example and test example can be found in branch "Wheres-Religion-Desktop-fall-2024"

This issue will help members new to the project to understand the work flow of the application and familiarize with the project.

### PR DESCRIPTION
**Pull request by jwang-101**
**Reference**
Fixes #150 

**What Changed?**
Upon completion of the intro.js tour, the user is automatically redirected to the "Create a Note" page.

**Why did it change?**
This was changed to give new users a smoother onboarding experience. New users may feel lost as to what to do in the Where's Religion website, but the tour helps guide them to pages of interest, by guiding them to the note creation page, which is a core part of this website.

**How was it changed?**
Using the oncomplete() function built-in to intro.js, once the tour is complete, the user will be automatically navigated to the create a note page.

### CODE REVIEW

File: app/lib/pages/map/page.tsx [hunk 1]
```diff
@@ -115,11 +118,24 @@ const Page = () => {
               intro: "Done for the day? Make sure to logout!"
             }
           ],
+
+          initialStep: pageTourStep ? parseInt(pageTourStep, 10) : 0, //save tour progress
```
Comment (amychen108) [inline app/lib/pages/map/page.tsx:122]: Same as first comment this is also a component of the saving aspect of intro.js which isn't complete yet right?

File: app/lib/pages/map/page.tsx [hunk 2]
```diff
@@ -115,11 +118,24 @@ const Page = () => {
...
+        intro.oncomplete(() => { //navigate to create a note when tour is over
+          window.location.href = "/lib/pages/notes";
+        });
+
         intro.start();
+
+        intro.onbeforeexit(() => {
+          const currentStep = intro._currentStep;  // Get the current step
+          sessionStorage.setItem('pageTourStep', currentStep); // Store the step in sessionStorage
+        });
```
Comment (amychen108) [inline app/lib/pages/map/page.tsx:137]: Same for these lines as above comments

File: app/lib/pages/map/page.tsx [hunk 3]
```diff
@@ -83,7 +83,10 @@ const Page = () => {
   const searchBarRef = useRef<HTMLDivElement | null>(null);
   const notesListRef= useRef<HTMLDivElement | null>(null);
   useEffect(() => {
-      const observer = new MutationObserver(() => {
+
+    const pageTourStep = sessionStorage.getItem('pageTourStep'); //remember where page tour was
```
Comment (amychen108) [inline app/lib/pages/map/page.tsx:87]: I don't think you need this/it is not completed yet.

**Review by BaloneyBoy97 — COMMENTED** [review]
Great work overall,
Product Mindset:
- added feature is aligned with the assigned task
- After the user clicks "Next" on the final tooltip on the map/page, they are redirected to the next page is in line with client priority
- code changes in pages/map is easy to understand and is in line with project architecture

Code Quality:
- the fundamentals of the code is sound
- performance is excellent, The "Skip Tour" button works across pages, allowing users to exit the tour at any time.
- Still need to create test cases and pass

Functionality verification:
- the code is running with no compile error
- the feature meet the task requirement.
    - need to finish tests
- Code is regression free atm

Focus On The Important Issues:
- Changes are focused on the priority issue

Good job Justin, with your second sprint issue. Keep up and work and remember to add test cases for your code before Monday to have it ready to merge

**Review by BaloneyBoy97 — APPROVED** [review]
Well Done Justin,

Feature meeting merge requirement

**Review by yashb196 — APPROVED** [review]

**Pull request merged**